### ISSUE TITLE
Bug in `size()`

### ISSUE THREAD

**Issue by laurimi** [artem-ogre/enumerate_combinations#1]
In the method [`size()`](https://github.com/artem-ogre/combinations_cpp/blob/304484c59fdc7faf54ed9384980bf1a7e6c1ac7e/combinations.hpp#L192) I think there is a bug.  I was not able to compile any code using `size()`.

I believe that the size of the vector needs to be referenced somewhere there instead of directly using `std::multiplies`.

To reproduce:
```
#include "combinations.hpp"
#include <iostream>
int main()
{
    std::vector<std::vector<double> > x(4, std::vector<double>(4, 42.0));
    auto ft = make_combinations(x);
    std::cout << ft.size() << std::endl;
    return 0;
}
```
Expected output: 256

Compilation error:

> /usr/include/c++/5/bits/stl_numeric.h: In instantiation of ‘_Tp std::accumulate(_InputIterator, _InputIterator, _Tp, _BinaryOperation) [with _InputIterator = __gnu_cxx::__normal_iterator<const std::vector<double>*, std::vector<std::vector<double> > >; _Tp = int; _BinaryOperation = std::multiplies<long unsigned int>]’:
> combinations.hpp:177:52:   required from ‘combinations<SetsIter>::size_type combinations<SetsIter>::size() const [with SetsIter = __gnu_cxx::__normal_iterator<const std::vector<double>*, std::vector<std::vector<double> > >; combinations<SetsIter>::size_type = long unsigned int]’
> text.cpp:14:26:   required from here
> /usr/include/c++/5/bits/stl_numeric.h:154:22: error: no match for call to ‘(std::multiplies<long unsigned int>) (int&, const std::vector<double>&)’
>   __init = __binary_op(__init, *__first);
>                       ^
> In file included from /usr/include/c++/5/string:48:0,
>                  from /usr/include/c++/5/bits/locale_classes.h:40,
>                  from /usr/include/c++/5/bits/ios_base.h:41,
>                  from /usr/include/c++/5/ios:42,
>                  from /usr/include/c++/5/ostream:38,
>                  from /usr/include/c++/5/iterator:64,
>                  from combinations.hpp:25,
>                  from text.cpp:1:
> /usr/include/c++/5/bits/stl_function.h:191:7: note: candidate: _Tp std::multiplies<_Tp>::operator()(const _Tp&, const _Tp&) const [with _Tp = long unsigned int]
>        operator()(const _Tp& __x, const _Tp& __y) const
>        ^
> /usr/include/c++/5/bits/stl_function.h:191:7: note:   no known conversion for argument 2 from ‘const std::vector<double>’ to ‘const long unsigned int&’

**Comment by artem-ogre** [comment]
Thanks for reporting!

Should be fixed with #2

@laurimi Please let me know if it worked for you.

**Comment by laurimi** [comment]
@artem-ogre works perfectly, thanks for the prompt fix!
